frontend/pages/policies/PolicyPage/PolicyPage.tsx

```diff
@@ -0,0 +1,232 @@
+
+// @ts-ignore
+import Fleet from "fleet"; // @ts-ignore
+import { AppContext } from "context/app";
+import { PolicyContext } from "context/policy";
+import { QUERIES_PAGE_STEPS, DEFAULT_POLICY } from "utilities/constants";
+import globalPoliciesAPI from "services/entities/global_policies"; // @ts-ignore
+import hostAPI from "services/entities/hosts"; // @ts-ignore
+import { IPolicyFormData, IPolicy } from "interfaces/policy";
+import { ITarget } from "interfaces/target";
+import { IHost } from "interfaces/host";
+
+import QuerySidePanel from "components/side_panels/QuerySidePanel";
+import QueryEditor from "pages/policies/PolicyPage/screens/QueryEditor";
+import SelectTargets from "pages/policies/PolicyPage/screens/SelectTargets";
+import RunQuery from "pages/policies/PolicyPage/screens/RunQuery";
+import ExternalURLIcon from "../../../../assets/images/icon-external-url-12x12@2x.png";
+
+interface IPolicyPageProps {
+  router: InjectedRouter;
+  params: Params;
+  location: any; // no type in react-router v3
+}
+
+interface IStoredPolicyResponse {
+  policy: IPolicy;
+}
+
+interface IHostResponse {
+  host: IHost;
+}
+
+const baseClass = "policy-page";
+
+const PolicyPage = ({
+  router,
+  params: { id: paramsPolicyId },
+  location: { query: URLQuerySearch },
+}: IPolicyPageProps): JSX.Element => {
+  const policyIdForEdit = paramsPolicyId ? parseInt(paramsPolicyId, 10) : null;
+  const {
+    isGlobalAdmin,
+    isGlobalMaintainer,
+    isAnyTeamMaintainerOrTeamAdmin,
+  } = useContext(AppContext);
+  const {
+    selectedOsqueryTable,
+    setSelectedOsqueryTable,
+    setLastEditedQueryName,
+    setLastEditedQueryDescription,
+    setLastEditedQueryBody,
+  } = useContext(PolicyContext);
+
+  const [step, setStep] = useState<string>(QUERIES_PAGE_STEPS[1]);
+  const [selectedTargets, setSelectedTargets] = useState<ITarget[]>([]);
+  const [isLiveQueryRunnable, setIsLiveQueryRunnable] = useState<boolean>(true);
+  const [isSidebarOpen, setIsSidebarOpen] = useState<boolean>(true);
+  const [
+    showOpenSchemaActionText,
+    setShowOpenSchemaActionText,
+  ] = useState<boolean>(false);
+
+  // disabled on page load so we can control the number of renders
+  // else it will re-populate the context on occasion
+  const {
+    isLoading: isStoredPolicyLoading,
+    data: storedPolicy,
+    error: storedPolicyError,
+    refetch: refetchStoredPolicy,
+  } = useQuery<IStoredPolicyResponse, Error, IPolicy>(
+    ["query", policyIdForEdit],
+    () => globalPoliciesAPI.load(policyIdForEdit as number),
+    {
+      enabled: false,
+      refetchOnWindowFocus: false,
+      select: (data: IStoredPolicyResponse) => data.policy,
+      onSuccess: (returnedQuery) => {
+        setLastEditedQueryName(returnedQuery.name);
+        setLastEditedQueryDescription(returnedQuery.description);
+        setLastEditedQueryBody(returnedQuery.query);
+      },
+    }
+  );
+
+  // if URL is like `/queries/1?host_ids=22`, add the host
```

Comment:
Should this comment be changed to `/policies/?host_ids=22`?

Reply:
```suggestion
  // if URL is like `/policies/1?host_ids=22`, add the host
```